executor/show.go

```diff
@@ -390,6 +390,10 @@ func (e *ShowExec) fetchShowColumns() error {
 		// The FULL keyword causes the output to include the column collation and comments,
 		// as well as the privileges you have for each column.
 		if e.Full {
+			switch desc.Type {
+			case "date", "datetime", "timestamp":
```

Comment:
I'm afraid all time type have this  problem. And note the decimal.
In mysql 
```sql
> create table t1 (a date,b datetime,c timestamp,d time, e year);
> show full columns from t1;
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+
| Field | Type      | Collation | Null | Key | Default           | Extra                       | Privileges                      | Comment |
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+
| a     | date      | <null>    | YES  |     | <null>            |                             | select,insert,update,references |         |
| b     | datetime  | <null>    | YES  |     | <null>            |                             | select,insert,update,references |         |
| c     | timestamp | <null>    | NO   |     | CURRENT_TIMESTAMP | on update CURRENT_TIMESTAMP | select,insert,update,references |         |
| d     | time      | <null>    | YES  |     | <null>            |                             | select,insert,update,references |         |
| e     | year(4)   | <null>    | YES  |     | <null>            |                             | select,insert,update,references |         |
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+

> create table t2 (a time(3),b datetime(3),c timestamp(3));
> show full columns from t2;
+-------+--------------+-----------+------+-----+----------------------+--------------------------------+---------------------------------+---------+
| Field | Type         | Collation | Null | Key | Default              | Extra                          | Privileges                      | Comment |
+-------+--------------+-----------+------+-----+----------------------+--------------------------------+---------------------------------+---------+
| a     | time(3)      | <null>    | YES  |     | <null>               |                                | select,insert,update,references |         |
| b     | datetime(3)  | <null>    | YES  |     | <null>               |                                | select,insert,update,references |         |
| c     | timestamp(3) | <null>    | NO   |     | CURRENT_TIMESTAMP(3) | on update CURRENT_TIMESTAMP(3) | select,insert,update,references |         |
+-------+--------------+-----------+------+-----+----------------------+--------------------------------+---------------------------------+---------+
```

Reply:
How about this now?
More test will be added into `tidb-test` when this PR is ready to merge.